R/add_smr_type.R

```diff
@@ -0,0 +1,75 @@
+#' Add smrtype variable based on record ID
+#'
+#' @param recid A vector of record IDs
+#' @param mpat A vector of mpat values
+#'
+#' @return A vector of SMR types
+#' @export
+#'
+#' @family Codes
+#'
+#' @examples
+#' x <- tibble::tibble(
+#'   recid = c("02B", "02B", "02B", "04B", "00B", "AE2", "PIS", "NRS"),
+#'   mpat = c("1", "2", "0", "", "", "", "", "")
+#' )
+#' add_smr_type(x)
+add_smr_type <- function(recid, mpat = NULL) {
+
+  # Situation where some recids are missing
+  if (any(is.na(recid)) & !all(is.na(recid))) {
+    cli::cli_inform(c("i" = "Some values of {.var recid} are {.val NA},
+                    please check this is populated throughout the data"))
+  }
+
+  # Situation where some recids are not in the accepted values
+  if (any(!(recid %in% c("02B", "04B", "00B", "AE2", "PIS", "NRS"))) &
+    !any(is.na(recid))) {
+    cli::cli_inform(c("i" = "One or more values of {.var recid} do not have an
+                   assignable {.var smrtype}"))
```

Comment:
Should this be a warning, rather than just a message? What would be the scenario where some non-handled recids were submitted and what should we do in that case? I'd almost be tempted to make it an error even.

```suggestion
    cli::cli_warn(c("i" = "One or more values of {.var recid} do not have an
                   assignable {.var smrtype}"))
```

Reply:
Changed this to `cli_warn`